### PR TITLE
demo nip04 with new example

### PR DESCRIPTION
**Pull request by futurepaul**
I think this exposes some ergonomic rough edges, but it does seem to be working!

Basically we need a nice way to ask Tags about what's up and I'm not sure what that is because there's so much optionality (and I don't think order is guaranteed either?

### CODE REVIEW

File: examples/dm.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    let (_alice_keypair, alice_pubkey, alice_sk) = gen_keys(ALICE_SK);
+    let (_bob_keypair, bob_pubkey, bob_sk) = gen_keys(BOB_SK);
+
+    let alice_to_bob = "Hey bob this is alice (ping)";
+    let bob_to_alice = "Hey alice this is bob (pong)";
+
+    let alice_encrypted_msg =
+        Event::new_encrypted_direct_msg(alice_sk, &bob_pubkey, alice_to_bob.clone()).as_json();
+
+    let subscribe_to_alice = format!("sub-key:{}", alice_pubkey);
```
Comment (emeceve) [inline examples/dm.rs]: Looking at Nostr specifications this is not the right way to make subscription. Is necessary build a JSON like `["REQ", <id>, <filter JSON>...]`. 

Related to filter, to sub for users messages is necessary use `authors` property of filter JSON. 

Complete Ex: `["REQ", "some_random_string", { "authors": ["alice_pk"] }]`

File: src/user.rs [hunk 2]
```diff
@@ -1 +1,17 @@
+use secp256k1::{
+    schnorrsig,
+    schnorrsig::{KeyPair, PublicKey},
+    Secp256k1, SecretKey,
+};
 
+use std::str::FromStr;
+
+/// Given a secret key return all the scep256k1 primitives
+pub fn gen_keys(sk: &str) -> (KeyPair, PublicKey, SecretKey) {
```
Comment (emeceve) [inline src/user.rs:10]: Useful method! Maybe worth make a struct for this.

**Review by emeceve — COMMENTED** [review]
I faced some incompatibilities while testing with relays using the [go](https://github.com/fiatjaf/nostr-relay-poc) implementation.

File: examples/dm.rs (hunk 1 above)
Comment (futurepaul) [reply examples/dm.rs]: fixed, don't know where I got this "sub-key" idea from

**Pull request merged**